gbpservice/nfp/service_plugins/firewall/nfp_fwaas_plugin.py

```diff
@@ -33,28 +39,53 @@ def _get_routers_for_create_firewall(self, tenant_id, context, firewall):
         # and not firewall db
         router_ids = firewall['firewall'].pop('router_ids', None)
         if router_ids == attr.ATTR_NOT_SPECIFIED:
-            # old semantics router-ids keyword not specified pick up
-            # all routers on tenant.
-            l3_plugin = manager.NeutronManager.get_service_plugins().get(
-                n_const.L3_ROUTER_NAT)
-            ctx = neutron_context.get_admin_context()
-            routers = l3_plugin.get_routers(ctx)
-            router_ids = [
-                router['id']
-                for router in routers
-                if router['tenant_id'] == tenant_id]
-            # validation can still fail this if there is another fw
-            # which is associated with one of these routers.
-            # Dont validate this condition
-            # self.validate_firewall_routers_not_in_use(context, router_ids)
-            return router_ids
-        else:
-            if not router_ids:
-                # This indicates that user specifies no routers.
-                return []
-            else:
-                # some router(s) provided.
-                # Dont validate this condition
-                # self.validate_firewall_routers_not_in_use(context,
-                # router_ids)
-                return router_ids
+            return tenant_id
+
+    def nfp_create_firewall(self, context, firewall, status=None):
```

Comment:
You are overwriting a method in super class. Name should match.
Remove nfp_ prefix. Just call create_firewall(...)